pandas/core/groupby/groupby.py

```diff
@@ -1120,12 +1117,14 @@ def _resolve_numeric_only(self, numeric_only: bool | lib.NoDefault) -> bool:
             if self.obj.ndim == 2:
                 # i.e. DataFrameGroupBy
                 numeric_only = True
+                obj = self._obj_with_exclusions
+                check = obj._get_numeric_data()
+                if len(obj.columns) and not len(check.columns) and not obj.empty:
+                    numeric_only = False
```

Comment:
and in a follow-up PR a warning goes here?

Reply:
> and in a follow-up PR a warning goes here?

I put a `# TODO` for this in https://github.com/pandas-dev/pandas/pull/43154/commits/45f54d6cb548828072fc2d5885687d4d4db5b712